### ISSUE TITLE
auto modification of tilt angles on import is unexpected

### ISSUE THREAD

**Issue by shahpnmlab** [warpem/warp#91]
After importing the tilt series, tilt-angle assignments to the frames in the tomostar are wonky. The command to produce this output was
```
WarpTools ts_import --mdocs mdoc --frameseries warp_frameseries --tilt_exposure 3.2 --min_intensity 0.3 --output tomostar
```

```
      │ File: tomostar/lam1_ts_015.tomostar
───────┼───────────────────────────────────────────────────
   1   │
   2   │ data_
   3   │
   4   │ loop_
   5   │ _wrpMovieName #1
   6   │ _wrpAngleTilt #2
   7   │ _wrpAxisAngle #3
   8   │ _wrpDose #4
   9   │ _wrpAverageIntensity #5
  10   │ _wrpMaskedFraction #6
  11   │   ../warp_frameseries/frame_001_00590_-66.0.eer   71.98  84.700  115.200005  9.266  0.000
  12   │   ../warp_frameseries/frame_001_00574_-63.0.eer   68.98  84.700         112  9.195  0.000
  13   │   ../warp_frameseries/frame_001_00526_-60.0.eer   65.99  84.700       102.4  9.203  0.000
  14   │   ../warp_frameseries/frame_001_00510_-57.0.eer   62.99  84.700   99.200005  9.297  0.000
  15   │   ../warp_frameseries/frame_001_00462_-54.0.eer   59.99  84.700        89.6  9.352  0.000
  16   │   ../warp_frameseries/frame_001_00446_-51.0.eer   56.99  84.700        86.4  9.352  0.000
  17   │   ../warp_frameseries/frame_001_00398_-48.0.eer   53.99  84.700        76.8  9.305  0.000
  18   │   ../warp_frameseries/frame_001_00382_-45.0.eer   50.99  84.700        73.6  9.422  0.000
  19   │   ../warp_frameseries/frame_001_00334_-42.0.eer   47.98  84.700          64  9.523  0.000
  20   │   ../warp_frameseries/frame_001_00318_-39.0.eer   44.99  84.700        60.8  9.547  0.000
  21   │   ../warp_frameseries/frame_001_00270_-36.0.eer   41.99  84.700        51.2  9.539  0.000
  22   │   ../warp_frameseries/frame_001_00254_-33.0.eer   38.99  84.700          48  9.609  0.000
  23   │   ../warp_frameseries/frame_001_00206_-30.0.eer   35.99  84.700        38.4  9.484  0.000
  24   │   ../warp_frameseries/frame_001_00190_-27.0.eer   32.99  84.700        35.2  9.445  0.000
  25   │   ../warp_frameseries/frame_001_00142_-24.0.eer   29.99  84.700        25.6  9.516  0.000
  26   │   ../warp_frameseries/frame_001_00126_-21.0.eer   26.99  84.700        22.4  9.586  0.000
  27   │   ../warp_frameseries/frame_001_00078_-18.0.eer   23.99  84.700        12.8  9.547  0.000
  28   │   ../warp_frameseries/frame_001_00062_-15.0.eer   20.99  84.700         9.6  9.492  0.000
  29   │   ../warp_frameseries/frame_001_00014_-12.0.eer   17.98  84.700           0  9.445  0.000
  30   │    ../warp_frameseries/frame_001_00030_-9.0.eer   14.99  84.700         3.2  9.570  0.000
  31   │    ../warp_frameseries/frame_001_00046_-6.0.eer   11.99  84.700         6.4  9.609  0.000
  32   │    ../warp_frameseries/frame_001_00094_-3.0.eer    8.99  84.700          16  9.586  0.000
  33   │    ../warp_frameseries/frame_001_00110_-0.0.eer    5.99  84.700        19.2  9.508  0.000
  34   │     ../warp_frameseries/frame_001_00158_3.0.eer    2.99  84.700   28.800001  9.578  0.000
  35   │     ../warp_frameseries/frame_001_00174_6.0.eer   -0.00  84.700          32  9.641  0.000
  36   │     ../warp_frameseries/frame_001_00222_9.0.eer   -3.01  84.700   41.600002  9.523  0.000
  37   │    ../warp_frameseries/frame_001_00238_12.0.eer   -6.00  84.700        44.8  9.406  0.000
  38   │    ../warp_frameseries/frame_001_00286_15.0.eer   -9.01  84.700        54.4  9.438  0.000
  39   │    ../warp_frameseries/frame_001_00302_18.0.eer  -12.01  84.700   57.600002  9.492  0.000
  40   │    ../warp_frameseries/frame_001_00350_21.0.eer  -15.01  84.700   67.200005  9.375  0.000
  41   │    ../warp_frameseries/frame_001_00366_24.0.eer  -18.01  84.700        70.4  9.453  0.000
  42   │    ../warp_frameseries/frame_001_00414_27.0.eer  -21.01  84.700          80  9.367  0.000
  43   │    ../warp_frameseries/frame_001_00430_30.0.eer  -24.00  84.700   83.200005  9.297  0.000
  44   │    ../warp_frameseries/frame_001_00478_33.0.eer  -27.00  84.700        92.8  9.406  0.000
  45   │    ../warp_frameseries/frame_001_00494_36.0.eer  -30.01  84.700          96  9.266  0.000
  46   │    ../warp_frameseries/frame_001_00542_39.0.eer  -33.01  84.700       105.6  9.156  0.000
  47   │    ../warp_frameseries/frame_001_00558_42.0.eer  -36.01  84.700       108.8  9.094  0.000

       │ File: tomostar/lam3_ts_007.tomostar
───────┼───────────────────────────────────────────────────
   1   │
   2   │ data_
   3   │
   4   │ loop_
   5   │ _wrpMovieName #1
   6   │ _wrpAngleTilt #2
   7   │ _wrpAxisAngle #3
   8   │ _wrpDose #4
   9   │ _wrpAverageIntensity #5
  10   │ _wrpMaskedFraction #6
  11   │   ../warp_frameseries/frame_001_01674_-57.0.eer   51.00  84.700  99.200005  8.258  0.000
  12   │   ../warp_frameseries/frame_001_01620_-54.0.eer   48.00  84.700       89.6  8.383  0.000
  13   │   ../warp_frameseries/frame_001_01602_-51.0.eer   45.00  84.700       86.4  8.453  0.000
  14   │   ../warp_frameseries/frame_001_01548_-48.0.eer   42.00  84.700       76.8  8.500  0.000
  15   │   ../warp_frameseries/frame_001_01530_-45.0.eer   38.99  84.700       73.6  8.672  0.000
  16   │   ../warp_frameseries/frame_001_01476_-42.0.eer   36.00  84.700         64  8.688  0.000
  17   │   ../warp_frameseries/frame_001_01458_-39.0.eer   32.99  84.700       60.8  8.742  0.000
  18   │   ../warp_frameseries/frame_001_01404_-36.0.eer   30.00  84.700       51.2  8.742  0.000
  19   │   ../warp_frameseries/frame_001_01386_-33.0.eer   27.00  84.700         48  8.711  0.000
  20   │   ../warp_frameseries/frame_001_01332_-30.0.eer   24.00  84.700       38.4  8.719  0.000
  21   │   ../warp_frameseries/frame_001_01314_-27.0.eer   21.00  84.700       35.2  8.750  0.000
  22   │   ../warp_frameseries/frame_001_01260_-24.0.eer   18.00  84.700       25.6  8.789  0.000
  23   │   ../warp_frameseries/frame_001_01242_-21.0.eer   15.00  84.700       22.4  8.812  0.000
  24   │   ../warp_frameseries/frame_001_01188_-18.0.eer   12.00  84.700       12.8  8.812  0.000
  25   │   ../warp_frameseries/frame_001_01170_-15.0.eer    9.00  84.700        9.6  8.773  0.000
  26   │   ../warp_frameseries/frame_001_01116_-12.0.eer    5.99  84.700          0  8.742  0.000
  27   │    ../warp_frameseries/frame_001_01134_-9.0.eer    3.00  84.700        3.2  8.852  0.000
  28   │    ../warp_frameseries/frame_001_01152_-6.0.eer   -0.00  84.700        6.4  8.852  0.000
  29   │    ../warp_frameseries/frame_001_01206_-3.0.eer   -3.00  84.700         16  8.805  0.000
  30   │    ../warp_frameseries/frame_001_01224_-0.0.eer   -6.00  84.700       19.2  8.812  0.000
  31   │     ../warp_frameseries/frame_001_01278_3.0.eer   -9.00  84.700  28.800001  8.781  0.000
  32   │     ../warp_frameseries/frame_001_01296_6.0.eer  -11.99  84.700         32  8.766  0.000
  33   │     ../warp_frameseries/frame_001_01350_9.0.eer  -14.99  84.700  41.600002  8.734  0.000
  34   │    ../warp_frameseries/frame_001_01368_12.0.eer  -18.00  84.700       44.8  8.719  0.000
  35   │    ../warp_frameseries/frame_001_01422_15.0.eer  -21.00  84.700       54.4  8.672  0.000
  36   │    ../warp_frameseries/frame_001_01440_18.0.eer  -24.00  84.700  57.600002  8.648  0.000
  37   │    ../warp_frameseries/frame_001_01494_21.0.eer  -26.98  84.700  67.200005  8.602  0.000
  38   │    ../warp_frameseries/frame_001_01512_24.0.eer  -29.99  84.700       70.4  8.500  0.000
  39   │    ../warp_frameseries/frame_001_01566_27.0.eer  -33.00  84.700         80  8.477  0.000
  40   │    ../warp_frameseries/frame_001_01584_30.0.eer  -35.99  84.700  83.200005  8.328  0.000
  41   │    ../warp_frameseries/frame_001_01638_33.0.eer  -39.00  84.700       92.8  8.336  0.000
  42   │    ../warp_frameseries/frame_001_01656_36.0.eer  -41.98  84.700         96  8.219  0.000
  43   │    ../warp_frameseries/frame_001_01710_39.0.eer  -45.00  84.700      105.6  8.070  0.000
  44   │    ../warp_frameseries/frame_001_01728_42.0.eer  -47.99  84.700      108.8  7.844  0.000
  ```

The mdoc files have the right angles associated with the right tilts. So I am not sure how I could have possibly mucked the inputs here.

**Comment by alisterburt** [comment]
could you provide an mdoc please @shahpnmlab ?

Here is the output for our test data where everything seems to be correct... note that tilt angles are inverted because of the different internal conventions in Warp


```txt
data_

loop_
_wrpMovieName #1
_wrpAngleTilt #2
_wrpAxisAngle #3
_wrpDose #4
_wrpAverageIntensity #5
_wrpMaskedFraction #6
  ../warp_frameseries/2Dvs3D_53-1_00041_-40.0_Jul31_11.05.12.tif   40.01  -93.500  105.600006  3.865  0.000
  ../warp_frameseries/2Dvs3D_53-1_00040_-38.0_Jul31_11.04.30.tif   38.00  -93.500   102.96001  3.981  0.000
  ../warp_frameseries/2Dvs3D_53-1_00037_-36.0_Jul31_11.02.04.tif   36.01  -93.500       95.04  3.948  0.000
```

**Comment by alisterburt** [comment]
It looks like you have a constant tilt angle offset, is this something you might have set during collection to flatten the lamella?

**Comment by shahpnmlab** [comment]
This is lamella data with a 12˚ offset... will share mdoc offline...

**Comment by shahpnmlab** [comment]
Closing because this is a feature and not a bug.

**Comment by alisterburt** [comment]
to clarify - @dtegunov told @shahpnmlab that the likely 'true zero degree image' is estimated from image intensities (highest median intensity at lowest tilt angle)

We might consider not doing this automatically and instead hiding it behind a flag at `ts_import`

**Comment by zxxshower** [comment]
Thanks for the information provided above!
But are there any follow-ups regarding the tilt angle problem? We encountered a similar issue, and it seems that by comparison, this modified tilt angle information results in worse alignment (we compared between two manual alignments in imod using the same parameters, the one processed by warp-tool with the different tilt angle from mdoc showed a higher alignment error score. But could be due to other reasons I might not have realized)...
And what does 'true zero degree image' mean? For me it is difficult to understand why the zero degree image might have lower intensity?

**Comment by alisterburt** [comment]
@zxxshower if the true 0 degree image is flat then the image intensity should be highest in that image as thickness increases with the tilt angle

Your note about the alignments is interesting - I'll make this functionality optional today ☺️

**Comment by alisterburt** [comment]
closed by 0dc29a3 and deploying to conda as `v2.0.0dev16`
